status-im/status-react
# Allow for env variables for configuring delete-message undo time limit for toasts

[comment]: # (Please replace ... with your information. Remove < and >)
[comment]: # (To auto-close issue on merge, please insert the related issue number after # i.e fixes #566)

fixes #19644 

### Summary

[comment]: # (Summarise the problem and how the pull request solves it)

* This PR attempts to the status-mobile to be configured with environment variables for configuring the time limit for displaying the undo toast after deleting a message.
* This PR adds two environment variables: `DELETE_MESSAGE_FOR_ME_UNDO_TIME_LIMIT` and `DELETE_MESSAGE_UNDO_TIME_LIMIT`.
  * `DELETE_MESSAGE_FOR_ME_UNDO_TIME_LIMIT` applies to deleting a message for the user in a chat
  * `DELETE_MESSAGE_UNDO_TIME_LIMIT` applies to deleting a message for everyone in a chat
* To use these environment variables we can define them inside the `.env` files for the status-mobile project. If we want to define these only for E2E tests, then we can define them inside the `.env.e2e` file inside the status-mobile project.

#### Platforms


- Android
- iOS

#### Areas that maybe impacted


##### Functional

- 1-1 chats

### Steps to test

- Define the environment variables inside the `.env` or `.env.e2e` files for testing.
- If you're building the application, then make sure to run `make clean` before building.
- Open the Status mobile app
- Open a 1-1 chat
- Send a message inside a 1-1 chat
- Long press on that message to open the action menu
- Choose "Delete for me" action or "Delete for everyone" action

status: ready
